thread_pthread.c

```diff
@@ -1964,9 +1965,10 @@ reserve_stack(volatile char *limit, size_t size)
 
 #undef ruby_init_stack
 void
-ruby_init_stack(volatile VALUE *addr)
+ruby_init_stack(volatile void *addr)
 {
     native_main_thread.id = pthread_self();
+    addr = asan_get_real_stack_addr((void *)addr);
```

Comment:
It's not obvious to me why we're using functions directly referencing asan in places where we don't know whether asan is available or enabled or not.

Reply:
the pattern for most of the existing ASAN macros is for them to be no-ops if ASAN is disabled, and to elide the checks where they're used. I guess the motivation for this is to avoid spraying a lot of conditional compilation checks all throughout the codebase, like:

```
#if RUBY_ASAN_ENABLED
    void *poisoned = asan_unpoison_object_temporary(val)
#endif
   /* use val */
#if RUBY_ASAN_ENABLED
  if (poisoned) { asan_poison_object(val); }
#endif
```

For these more specialised macros only used in a couple of places though, perhaps it's better to be explicit and put the check here.